test_malformed_csp fail because #254 fixes accept named hostname

``` bash

======================================================================
FAIL: test_malformed_csp (functional.plugins.test_csp.TestCSPPlugin)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/vagrant/minion/minion-backend/tests/functional/plugins/test_csp.py", line 183, in test_malformed_csp
    expected={'value': 'img-src', 'name': 'default-src', 'hname': 'x-content-security-policy'})
  File "/home/vagrant/minion/minion-backend/tests/functional/plugins/base.py", line 70, in validate_plugin
    return validator(runner_resp, request_resp, expected=expected, expectation=expectation)
  File "/home/vagrant/minion/minion-backend/tests/functional/plugins/test_csp.py", line 122, in validate_csp
    value=expected['value'], name=expected['name']), runner_resp[1]['data']['Description'])
AssertionError: 'Malformed CSP header set: img-src does not seem like a valid source expression for default-src' != u'Malformed CSP header set: When * is present, other values cannot co-exist with *'
```

This is because CSP accepts hostname like `localhost`. This means `script-src` can be a valid hostname.

--- a/tests/functional/plugins/test_csp.py
+++ b/tests/functional/plugins/test_csp.py
@@ -27,7 +27,8 @@ RULES = {'DEFAULT-SRC': "default-src 'self'",
          "EVAL": "script-src 'self' 'unsafe-eval' https://mydomain.com",
          "INLINE": "script-src 'self' 'unsafe-inline' https://mydomain.com",
          "BLOB": "default-src 'self' *.mega.co.nz http://*.mega.co.nz;" + 
-                 "script-src 'self' mega.co.nz data: blob:;"
+                 "script-src 'self' mega.co.nz data: blob:;",
+         "MALFORMED": "default-src 'selff'",
 }
 
 @test_app.route('/no-csp')
@@ -53,8 +54,7 @@ def default_src_self_trusted_subdomain_all_img_some_media_some_script(type):
 
 @test_app.route('/malformed-csp')
 def malformed_csp():
-    mf = RULES['DEFAULT-SRC-DOMAIN-SUB-DOMAIN-IMG-MEDIA-SCRIPT'].replace(';', '')
-    return _make_res('x', mf)
+    return _make_res('c', RULES['MALFORMED'])
 
 @test_app.route('/eval-csp')
 def eval_csp():
@@ -179,7 +179,7 @@ Content-Security-Policy to secure your site.', runner_resp[1]['data']['Descripti
     def test_malformed_csp(self):
         api_name = '/malformed-csp'
         self.validate_plugin(api_name, self.validate_csp, expectation='INVALID',
-                expected={'value': 'img-src', 'name': 'default-src', 'hname': 'x-content-security-policy'})
+                expected={'value': "'selff'", 'name': 'default-src', 'hname': 'content-security-policy'})
     def test_eval_csp(self):
         api_name = '/eval-csp'
         self.validate_plugin(api_name, self.validate_csp, expectation='UNSAFE-EVAL')

